src/bootstrap/lib.rs

```diff
@@ -65,9 +65,15 @@
 //! also check out the `src/bootstrap/README.md` file for more information.
 
 #![deny(warnings)]
+#![feature(associated_consts)]
+#![feature(core_intrinsics)]
```

Comment:
I reckon there are still some subtleties here with choosing the correct map, but I think I can see some options. One of my experiments is almost done so I'll make a PR against your repo and link to it from here.